### ISSUE TITLE
Huh?

### ISSUE THREAD

**Issue by hueniverse** [gr2m/hapi-cors-headers#24]
Can someone explain the purpose of this module? I'm confused.

**Comment by gr2m** [comment]
Hi Eran 👋 

It’s been a while since I used it myself. Back when we created it to use it with Hoodie, we needed a CORS API which works universally for all origins and accepts authentication (`Access-Control-Allow-Credentials: true`). If I recall correctly, just setting `Access-Control-Allow-Origin: *` is not allowed in combination with `Access-Control-Allow-Credentials: true`, so we set it dynamically to the request origin (see [/index.js#L12](https://github.com/gr2m/hapi-cors-headers/blob/0d94d14658ec378575640c55b7f57fd4e93d04a7/index.js#L12)).

I’ve never updated the module to the latest Hapi though since I stopped using it myself. 

Does Hapi support this behavior out of the box now?

**Comment by hueniverse** [comment]
So this basically enables CORS while also disabling all of its protections?

**Comment by gr2m** [comment]
Yes, you should not use it if your server uses cookies for authentication. I should probably put a big warning in the README 🤔
